docker/entrypoint.sh

```diff
@@ -9,4 +11,23 @@ if [ -n "$METADATA_AGENT_URL" ]; then
     /etc/google-fluentd/google-fluentd.conf
 fi
 
-/usr/sbin/google-fluentd "$@"
+trap 'kill $pid' TERM INT
+
+# first arg is a flag.
+if [ "${1:0:1}" = '-' ]; then
+  /usr/sbin/google-fluentd "$@" &
+else
+  exec "$@" &
```

Comment:
Why is this needed?

Reply:
Same question.

Reply:
This is a typical thing to see in docker entrypoints. This allows a user to run their own commands, such as "bash".

If the command looks something like
```
docker run -d logging-agent -o /var/log/google-fluentd.log
```
it will use the first condition, passing the flags to the google-fluentd process.

If the command looks like 
```
docker run -d logging-agent bash
```

It will exec "bash".

If the command looks like 
```
docker run -d logging-agent
```

It will exec the CMD defined in the dockerfile, which is google-fluentd.

Reply:
Makes sense. Thanks for explaining. Do you think it's worth commenting it with a brief clarification? Just for the benefit of people who are newbies for the Docker world.